### PR TITLE
Upgraded Ruby to 3.0.2 and Rails to 6.1.4.1

### LINKED ISSUE #652: Try to upgrade Ruby to 3.0.2 and upgrade Rails to 6.1.4.1

### PR DESCRIPTION
**Pull request by yedhink**
Fixes #652

### CODE REVIEW

**Comment by yedhink** [comment]
@unnitallman @ghousemohamed 
Requesting reviews from both of you given that this is kind of like a major upgrade.

Ghouse: Please verify the docker config in this PR.

Once done, just approve the review or comment here. Will merge this.

**Review by ghousemohamed — APPROVED** [review]

File: config/initializers/backtrace_silencers.rb [hunk 1]
```diff
@@ -3,7 +3,8 @@
...
 
 # You can add backtrace silencers for libraries that you're using but don't wish to see in your backtraces.
-# Rails.backtrace_cleaner.add_silencer { |line| line =~ /my_noisy_library/ }
+# Rails.backtrace_cleaner.add_silencer { |line| /my_noisy_library/.match?(line) }
 
-# You can also remove all the silencers if you're trying to debug a problem that might stem from framework code.
-# Rails.backtrace_cleaner.remove_silencers!
+# You can also remove all the silencers if you're trying to debug a problem that might stem from framework code
+# by setting BACKTRACE=1 before calling your invocation, like "BACKTRACE=1 ./bin/rails runner 'MyClass.perform'".
+Rails.backtrace_cleaner.remove_silencers! if ENV["BACKTRACE"]
```
Comment (unnitallman) [inline config/initializers/backtrace_silencers.rb:10]: Instead of `BACKTRACE`, make it a bit more descriptive by saying `ENABLE_BACKTRACE`

Comment (yedhink) [reply config/initializers/backtrace_silencers.rb:10]: @unnitallman This was added by Rails itself. I kept it as the default. All the changes made to `config/**/*.rb` is done by Rails itself.

Is it a good idea to change the name given that with next upgrade Rails will try to re-add this? 

Let me know your thoughts.

Comment (unnitallman) [reply config/initializers/backtrace_silencers.rb:10]: If its something that was auto-generated, let us keep that way.

**Review by unnitallman — APPROVED** [review]
Looks good to me

**Pull request merged**